source.go

```diff
@@ -0,0 +1,156 @@
+package conf
+
+import (
+	"bytes"
+	"flag"
+	"html/template"
+	"reflect"
+	"strings"
+)
+
+// Source is the interface that allow new types to be plugged into a loader to
+// make it possible to load configuration from new places.
+//
+// When the configuration is loaded the Load method of each source that was set
+// on a loader is called with a pointer to an in-memory structure representing
+// the configuration object passed as dst.
+type Source interface {
+	Load(dst interface{}) error
+}
+
+// FlagSource is a special case of a source that receives a configuration value
```

Comment:
That's really not how it would work.

The config struct has just your configuration, the custom sources you set are just different places to get the configuration from. And the default loader has been reimplemented with a file and env sources to mirror the historical behavior using the new concepts (as a proof that it's working).

Nothing is fancy here, the source just injects a flag is the set of flags supported by the loader so it we can pass the path to a configuration file to load the config from (the way it used to work).